app/cookies/CookieType.scala

```diff
@@ -0,0 +1,40 @@
+package cookies
+
+import play.api.mvc.Cookie
+
+abstract class CookieType {
+  val name: String
+  val maxAge: Option[Int] = None
+  val path: String = "/"
+  val domain: Option[String] = None
+  val secure: Boolean = false
+  val httpOnly: Boolean = false
+}
+
+/**
+ * Represents data of type [[A]] being able to set the value for a given cookie type [[C]].
+ */
+trait CookieValue[A, C <: CookieType] {
```

Comment:
As we talked about, I think as much as I like that nifty implicit conversion, we may as well directly go for a string at the time we set the cookie. That will simplify the code and make it more straightforward. (less beautiful though 😃  )